src/simulator/cube.py

```diff
@@ -11,29 +10,30 @@ def __init__(self, length: float, height: float, width: float):
         self.height = height
         self.width = width
 
-    def bounding_box(self) -> Box:
+    def bounding_box(self, position: Position) -> Box:
         """Returns smallest enclosing upright Box
         """
-
-        # TODO: The following calculation is only for an upright cube
-        # dimension calculations
-        length = self.shape.length
-        height = self.shape.height
-        width = self.shape.width
-
-        # position calculations
-        x = self.position.x
-        y = self.position.y
-        z = self.position.z
-        phi = self.position.phi
-        theta = self.position.theta
-
-        # assign return Arguments
-        bb_cube = Cube(length, height, width)
-        bb_position = Position(x, y, z, phi, theta)
-
-        # return Box(1, bb_cube, bb_position)
-        pass
+        l = self.length
+        w = self.width
+        h = self.height
+        points = [[l / 2, w / 2, h / 2], [-l / 2, w / 2, h / 2],
```

Comment:
do we actually need all 8 points? for cylinder we're taking 4 points on top disk.